onnxruntime/core/framework/kernel_registry_manager.cc

```diff
@@ -51,6 +51,34 @@ void KernelRegistryManager::RegisterKernelRegistry(std::shared_ptr<KernelRegistr
   }
   custom_kernel_registries_.push_front(kernel_registry);
 }
+
+// Layout tranformer can add new nodes to the graph.
+// Since layout transformation can happen in an extended build, we need a way to
+// get the hashes for these nodes if they are not picked up and compiled by NNAPI
+// or some other compile based EP. Since the infrastructure as well as op_schema required to generate
+// these hashes is not available in an extended minimal build, we maintain a static map of
+// nodes to hash value. This hash value can then be used to retireive the kernel for the given op.
+static const std::unordered_map<std::string, HashValue> static_kernel_hashes{
```

Comment:
I guess it depends on how large the list is and how different it may be from model to model. If it's large or fairly static, then perhaps putting it a global list makes more sense than putting it in each model.
Could you elaborate on the concern with how manageable a global list is?

Reply:
I don't think the list should be model specific - more the finite set of things required for specific runtime optimizations which may or may not apply to a model. 

Mainly a concern about what we have to do if we have to add a kernel that has a typed registration to the list. Does that blow out the number of entries in the list with an entry per type per opset, making it hard to maintain.

Reply:
Perhaps we can check this list from a unit test to ease maintenance.